engine/verification/fetcher/engine.go

```diff
@@ -97,7 +97,9 @@ func (e *Engine) Ready() <-chan struct{} {
 	if e.chunkConsumerNotifier == nil {
 		e.log.Fatal().Msg("missing chunk consumer notifier callback in verification fetcher engine")
 	}
-	return e.unit.Ready()
+	return e.unit.Ready(func() {
+		<-e.requester.Ready()
```

Comment:
Should do `<-e.requester.Done()` in `Done` method so teardown is handled as well

Reply:
fixed in https://github.com/onflow/flow-go/pull/728/commits/1cd712e2d34961c14f2ad0797c6051028bd4e6f1